pandas/core/algorithms.py

```diff
@@ -1857,7 +1857,7 @@ def diff(arr, n: int, axis: int = 0, stacklevel=3):
                 "dtype lost in 'diff()'. In the future this will raise a "
                 "TypeError. Convert to a suitable dtype prior to calling 'diff'.",
                 FutureWarning,
-                stacklevel=stacklevel,
+                stacklevel=stacklevel + 1,
```

Comment:
Can you change the value directly in line 1817? From 3 to 4.

Reply:
After merging with master, tests pass even with stacklevel=3, so I changed it.